File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingCalendarColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TrainingCalendarRow } from "./TrainingCalendarRow";

export interface TrainingCalendarColumns {
    TrainingCalendarId: Column<TrainingCalendarRow>;
    Name: Column<TrainingCalendarRow>;
    TrainingTermName: Column<TrainingCalendarRow>;
    StartDate: Column<TrainingCalendarRow>;
    EndDate: Column<TrainingCalendarRow>;
    IsEnabled: Column<TrainingCalendarRow>;
    IsActive: Column<TrainingCalendarRow>;
}

export class TrainingCalendarColumns extends ColumnsBase<TrainingCalendarRow> {
    static readonly columnsKey = 'Administration.TrainingCalendar';
    static readonly Fields = fieldsProxy<TrainingCalendarColumns>();
}

File: QadeerApp.Web/Modules/Administration/TrainingCalendar/TrainingCalendarColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.TrainingCalendar")]
[BasedOnRow(typeof(TrainingCalendarRow), CheckNames = true)]
public class TrainingCalendarColumns
{
    [EditLink, AlignRight]
    public int TrainingCalendarId { get; set; }

    [EditLink, Width(200)]
    public string Name { get; set; }

    [Width(180)]
    public string TrainingTermName { get; set; }

    [Width(120), Sortable(true)]
    public DateTime StartDate { get; set; }

    [Width(120), Sortable(true)]
    public DateTime EndDate { get; set; }

    [Width(90)]
    public bool IsEnabled { get; set; }

    [Width(90)]
    public bool IsActive { get; set; }
}
